### PR TITLE
[ADF-5379] Add a way to disable WS notifications

### PR DESCRIPTION
**Pull request by davidcanonieto**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/ADF-5379

**What is the new behaviour?**
Add a way to disable websockets from the app.config.json so that we can stop them when running e2e tests and improve performance.

Two ways have been added:

- Directly from the component's input `enableNotifications`
- From your app.config.json 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/src/app.config.json [hunk 1]
```diff
@@ -1009,6 +1009,9 @@
     "expandedSidenav": true,
     "preserveState": true
   },
+  "adf-cloud-task-filters": {
+    "ws-notifications": true
```
Comment (mauriziovitale) [inline demo-shell/src/app.config.json:1013]: I think that we should have a ws-notification true false across components. right now task list is the only one that uses it but as next task we will have the same for process.

Comment (mauriziovitale) [reply demo-shell/src/app.config.json:1013]: @eromano @popovicsandras @DenysVuika what do u guys think?

File: lib/process-services-cloud/src/lib/task/task-filters/components/task-filters-cloud.component.ts [hunk 2]
```diff
@@ -48,12 +55,18 @@ export class TaskFiltersCloudComponent extends BaseTaskFiltersCloudComponent imp
     currentFilter: TaskFilterCloudModel;
 
     constructor(private taskFilterCloudService: TaskFilterCloudService,
-                private translationService: TranslationService) {
+                private translationService: TranslationService,
+                appConfigService: AppConfigService) {
         super();
+        if (this.enableNotifications === undefined) {
+            this.enableNotifications = appConfigService.get(TaskFiltersCloudComponent.WS_NOTIFICATIONS_KEY, false);
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/task/task-filters/components/task-filters-cloud.component.ts:62]: move this to the init

File: demo-shell/src/app.config.json (hunk 1 above)
Comment (DenysVuika) [reply demo-shell/src/app.config.json:1013]: what about centralizing around the "ws-notifications" as a top feature with some params, like:

```
{
  "notifications": {
    "task-filters": true,
    "process-filters": true
}
```

it's more like iOS/Android notification center settings as for me.

Also, we should not embed the technology stack into names, ws (web sockets) should not be in the name, what if we use different stack in the future... should be just "notifications" as for me

Comment (davidcanonieto) [reply demo-shell/src/app.config.json]: I like that @DenysVuika, It'll allow for future expandability

Comment (mauriziovitale) [reply demo-shell/src/app.config.json]: I agree to have a general notification property but I don't see to point of having the ability to turn one on and another off. We should probably call the property _**web-socket-notification**_

File: lib/process-services-cloud/src/lib/task/task-filters/components/task-filters-cloud.component.ts [hunk 3]
```diff
@@ -31,6 +31,13 @@ import { TaskCloudEngineEvent } from '../../../models/engine-event-cloud.model';
     styleUrls: ['base-task-filters-cloud.component.scss']
 })
 export class TaskFiltersCloudComponent extends BaseTaskFiltersCloudComponent implements OnInit, OnChanges {
+
+    static WS_NOTIFICATIONS_KEY = 'notifications.task-filters';
+
+    /** Enable Bubble Notifications on Task Filter Count */
+    @Input()
+    enableNotifications: boolean;
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/task/task-filters/components/task-filters-cloud.component.ts:39]: Do we really need this input? From my point of view, either is the parent to check the app config.
```this.enableNotifications = this.appConfigService.get(TaskFiltersCloudComponent.WS_NOTIFICATIONS_KEY, false);```
and pass the information down or the component itself

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
comment

File: demo-shell/src/app/services/app-notifications.service.ts [hunk 4]
```diff
@@ -54,7 +54,7 @@ export class AppNotificationsService {
         private alfrescoApiService: AlfrescoApiService
     ) {
         this.alfrescoApiService.alfrescoApiInitialized.subscribe(() => {
-            if (this.isProcessServicesEnabled()) {
+            if (this.isProcessServicesEnabled() && this.areWSNotificationsEnabled()) {
```
Comment (DenysVuika) [inline demo-shell/src/app/services/app-notifications.service.ts:57]: can we reduce technology from the naming and have it called `notificationsEnabled` function?

File: demo-shell/src/app/services/app-notifications.service.ts [hunk 5]
```diff
@@ -76,10 +76,14 @@ export class AppNotificationsService {
         });
     }
 
-    private isProcessServicesEnabled() {
+    private isProcessServicesEnabled(): boolean {
         return this.authenticationService.isLoggedIn() && (this.authenticationService.isBPMProvider() || this.authenticationService.isALLProvider());
     }
 
+    private areWSNotificationsEnabled(): boolean {
```
Comment (DenysVuika) [inline demo-shell/src/app/services/app-notifications.service.ts:83]: I suggest `notificationsEnabled` to make it more generic

**Review by DenysVuika — COMMENTED** [review]
Looks good, but check the comments

**Review by mauriziovitale — APPROVED** [review]

**Review by eromano — APPROVED** [review]

**Pull request merged**